utils/github/query.py

```diff
@@ -0,0 +1,179 @@
+# -*- coding: utf-8 -*-
+
+import requests
+
+
+class Query:
+    def __init__(self, token, max_page_size=100, pull_request_page_size=5):
+        self._token = token
+        self._max_page_size = max_page_size
+        self._pull_request_page_size = pull_request_page_size
+
+    ''' Get all pull-requests associated with at least one commit in range (until, default head],
+        where "default head" is a head of default repository branch.
+    '''
+    def get_pull_requests(self, until):
+        pull_requests = []
+        query = Query._FIRST.format(max_page_size=self._max_page_size, pull_request_page_size=self._pull_request_page_size)
+        not_end = True
+
+        default_branch_name = self.get_default_branch()
+
+        while not_end:
+            result = self._run(query)['data']['repository']['defaultBranchRef']['target']['history']
+            not_end = result['pageInfo']['hasNextPage']
+
+            for commit in result['edges']:
+                node = commit['node']
+
+                if str(node['oid']) == str(until):
+                    not_end = False
+                    break
+
+                # TODO: fetch all pull-requests that were merged in a single commit.
+                assert node['associatedPullRequests']['totalCount'] <= self._pull_request_page_size
```

Comment:
Maybe add assertion message? And why we check it?

Reply:
We check it, until TODO above is fixed. Otherwise, we can potentially miss the proper PR, if there are more, but we read only the first page.